src/awkward/highlevel.py

```diff
@@ -1387,12 +1387,62 @@ def _repr(self, limit_cols):
 
         return f"<{pytype}{valuestr}{axisstr} type={typestr}>"
 
+    def _show_rows(
```

Comment:
How different are the two `_show_rows` functions? Is one under `Array` and the other under `Record`?

Reply:
they're exactly the same, in fact many are the same between `Array` and `Record`, e.g. `show` and many properties such as `nbytes`, `layout`, `behavior`, `named_axis`, `positional_axis`, and also some functions such as `to_list()`. They mainly differ in their docstring though.

Reply:
if we would have a `ak.highlevel.HighLevelArray` base class these things could live there. It would maybe also simplify checks like:
`isinstance(array, (ak.Array, ak.Record))` -> `isinstance(array, ak.highlevel.HighLevelArray)`

Reply:
I moved `self._show_rows` into a helper function to avoid code duplication :+1: